boring-sys/patches/boring-pq.patch

```diff
@@ -20,43 +20,55 @@ This patch adds:
    key agreement should only be used for testing: to see if the smaller
    keyshare makes a difference.
 
+4. Supportfor IPDWing under codepoint 0xfe41. This key agreement
+   is a preliminary version of X-Wing using the initial public draft
+   of ML-KEM. It should not be used.
+
 The patch also replaces Google's implementation of Kyber, by the
 portable reference implementation, so as to support Kyber512.
 
 Cf RTG-2076 RTG-2051 RTG-2508 RTG-2707 RTG-2607
 ---
- BUILD.generated.bzl               |    5 +-
+ BUILD.generated.bzl               |    7 +-
  BUILD.generated_tests.bzl         |    4 -
- CMakeLists.txt                    |    4 +-
- sources.json                      |    9 +-
- src/crypto/CMakeLists.txt         |    5 +-
+ CMakeLists.txt                    |    6 +-
+ sources.json                      |   11 +-
+ src/crypto/CMakeLists.txt         |    7 +-
+ src/crypto/kyber/fips202.c        |  699 +++++++
+ src/crypto/kyber/fips202.h        |   29 +
  src/crypto/kyber/internal.h       |   91 -
+ src/crypto/kyber/ipdwing.c        |  110 ++
  src/crypto/kyber/keccak.c         |  204 --
- src/crypto/kyber/kyber.c          | 2865 ++++++++++++++++++++-------
+ src/crypto/kyber/kyber.c          | 2319 +++++++++++++++-------
```

Comment:
Reviewer note: Most of the changes here are moving SHA-3 out of this file into fips202.c. Also, some functions have been added for SHAKE256 (needed for IPD); and some unused functions have been pruned.